`Assessment`: Show point deduction directly in the text feedback component

maximiliansoelch opened the pull request:
<!-- Thanks for contributing to Artemis! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->
<!-- If your pull request is not ready for review yet, create a draft pull request! -->

### Checklist
#### General
<!-- You only need to choose one of the first two check items: Generally, test on the test servers. -->
<!-- If it's only a small change, testing it locally is acceptable and you may remove the first checkmark. If you are unsure, please test on the test servers. -->
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->
Fixes #5114 

### Description
<!-- Describe your changes in detail -->
Added the points assigned to a text feedback to the beginning of the associated feedback.
Renamed `credits` to `points` in the English translation. 
In addition, the points are now also getting localized based on the set language in Artemis.

### Steps for Testing
<!-- Please describe in detail how the reviewer can test your changes. -->
Prerequisites:
- 1 Instructor
- 1 Student
- 1 Text Exercise with Feedback

1. Submit a text as a student
2. Assess the submission and use a negative as well as positive inline feedback
3. As a student, you should see the points directly on the right side of the inline feedback
4. Change the language and make sure the points get localized as well

### Review Progress
<!-- Each Pull Request should be reviewed by at least two other developers. The code as well as the functionality (= manual test) needs to be reviewed. -->
<!-- The reviewer or author check the following boxes depending on what was reviewed or tested. All boxes should be checked before merge. -->
<!-- You can add additional checkboxes if it makes sense to only review parts of the code or functionality. -->
<!-- When changes are pushed, uncheck the affected boxes. (Not all changes require full re-reviews.) -->

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage
<!-- Please add the test coverages for all changed files here. You can see this when executing the tests locally (see build.gradle and package.json) or when looking into the corresponding Bamboo build plan. -->
<!-- Lines are the main reference but a significantly lower branch percentage can indicate missing edge cases in the tests. -->
| Class/File | Branch | Line |
|------------|-------:|-----:|
| text-result.component.ts | 70.73% | 95.16% |
| additional-feedback.component.ts | 78.57% | 100% |

### Screenshots
<!-- Add screenshots to demonstrate the changes in the UI. -->
<!-- Create a GIF file from a screen recording in a docker container https://toub.es/2017/09/11/high-quality-gif-with-ffmpeg-and-docker/ -->

![Bildschirmfoto 2022-05-20 um 23 04 23](https://user-images.githubusercontent.com/9954674/169610990-0b1b6fea-bef5-429f-8bf8-a8b711d0c72d.png)

![Bildschirmfoto 2022-05-20 um 23 03 57](https://user-images.githubusercontent.com/9954674/169610995-3ee9ec17-409f-4ab5-bc29-04dc5ff37271.png)

bassner reviewed (changes requested):
Thank you for that!

I have two suggestions:

1. With longer feedback texts, it doesn't look that good anymore. Maybe we could limit the width of the feedback text so it doesn't actually cross the red line, and move the score up to the top or the middle (vertically)?
![image](https://user-images.githubusercontent.com/23171488/169068725-9f3e5ee8-09d2-4356-ad28-5fbccefc12c0.png)


2. Apply the same format for the "additional feedback" boxes as they currently still have the point value on the left side
![image](https://user-images.githubusercontent.com/23171488/169068303-93dbc0d2-0d2d-4fda-8022-fb98c1494510.png)

maximiliansoelch commented:
@bassner I like your suggestions and will try to implement them

maximiliansoelch commented:
@bassner I addressed your suggestions a bit differently:
Instead of having the points on the right side, I placed them to the beginning of the associated feedback text (like it is currently displayed for the additional feedback) and additionally underlined the points. I think, this makes it easier to distinguish between assigned points and textual feedback.

bassner commented on src/main/webapp/app/exercises/text/participate/text-result/text-result.component.html:
```diff
@@ -5,11 +5,19 @@
         <div *ngIf="block.feedback && block.feedback.isSubsequent == undefined" class="feedback-text mt-1" [ngClass]="block.feedbackCssClass">
-            <b><fa-icon *ngIf="block.circleIcon" [icon]="block.circleIcon"></fa-icon>&nbsp;<span jhiTranslate="artemisApp.assessment.detail.feedback">Feedback</span>:</b>
+            <b
+                ><fa-icon *ngIf="block.circleIcon" [icon]="block.circleIcon"></fa-icon>&nbsp;<u
+                    ><span>{{ creditsTranslationForTextResultBlock(block) }}</span></u
+                >:</b
+            >
             <span [innerHTML]="buildFeedbackTextForReview(block.feedback)" id="text-feedback-{{ i }}"></span>
         </div>
         <div *ngIf="block.feedback && block.feedback.isSubsequent" class="feedback-text mt-1 alert alert-secondary">
-            <b><fa-icon *ngIf="block.circleIcon" [icon]="block.circleIcon"></fa-icon>&nbsp;<span jhiTranslate="artemisApp.assessment.detail.feedback">Feedback</span>:</b>
+            <b
+                ><fa-icon *ngIf="block.circleIcon" [icon]="block.circleIcon"></fa-icon>&nbsp;<u
+                    ><span>{{ creditsTranslationForTextResultBlock(block) }}</span></u
+                >:</b
+            >
```
I have a few thoughts about this, let me know what you think :)

- I'd suggest to not underline it. I tried clicking / hovering on it! :D
From the spec:
```The default rendering of the <u> element in visual presentations clashes with the conventional rendering of hyperlinks (underlining). Authors are encouraged to avoid using the <u> element where it could be confused for a hyperlink.```
- The indentation and newline use of these lines look a bit weird :D can we fix that? E.g. currently there is a newline between ``<b`` and ``>``, for example :)

e.g. we could just go with
```html
<b>
    <fa-icon>...</fa-icon>
    &nbsp; {{ creditsTranslationForTextResultBlock(block) }}:
</b>
```

maximiliansoelch replied:
Thanks for the review! I implemented your suggestions.

bassner reviewed (changes requested):
I like this version, too! Looks very consistent now in both areas. 
Just one last comment below ... :)

joaquinmurcia reviewed (dismissed):
Tested on TS1. Works as expected.
![image](https://user-images.githubusercontent.com/47792433/169565184-07ea8759-5d09-46ca-ad4d-ee3094e305b9.png)

nairaarnez reviewed (approved):
Tested in ts2, the text feedback component works as expected! I only have one comment, and it's from the student perspective. There is an inconsistency in the result view. Here I just graded the assignment without inline feedback. The exercise is already graded, but the "not graded" still remains. 
![image](https://user-images.githubusercontent.com/104493356/169576599-98aa1920-fe96-46b1-acc0-7c2adc9bb2e0.png)

maximiliansoelch commented:
@nairaarnez Thanks for testing and reporting this bug 👍
But please create a new issue for this bug as it is not within the scope of this PR.

nadinesamyy reviewed (approved):
tested on ts2, works as expected and as shown in screenshots

xlieA reviewed (approved):
Tested on ts1, followed the steps for testing and everything looks great.

TheTrueVester reviewed (approved):
Tested on ts1. Everything worked as described.

babyygemperor reviewed (approved):
Tested on ts2, works as described 👍🏼
